### DISCUSSION TITLE
Shared `Tab` component

### DISCUSSION THREAD

**Discussion by oliverlloyd** [guardian/source#815] [Ideas]
# `Tab`
A component to centralise the design for a Guardian Tab

Each of these pages have 'tabs' but with varying styles and features. We might not want _all_ of these pages to have their tabs be identical but perhaps there's some crossover making a shared component worth it?

Tabs are often used as one of the core methods for navigation so by centralising them we can bring accessibility benefits.

https://manage.theguardian.com/help-centre
https://profile.theguardian.com/reset
https://manage.theguardian.com/billing
https://www.theguardian.com/about/history
https://www.theguardian.com/uk

![Screenshot 2021-05-12 at 12 13 42](https://user-images.githubusercontent.com/1336821/117966175-87470700-b31b-11eb-8c4e-ab628684fb95.jpg)
![Screenshot 2021-05-12 at 12 13 29](https://user-images.githubusercontent.com/1336821/117966177-87df9d80-b31b-11eb-9416-adb4c1efc575.jpg)
![Screenshot 2021-05-12 at 12 13 18](https://user-images.githubusercontent.com/1336821/117966180-88783400-b31b-11eb-88b3-efa7b23cc76f.jpg)
![Screenshot 2021-05-12 at 12 12 50](https://user-images.githubusercontent.com/1336821/117966186-8910ca80-b31b-11eb-8345-c4db03e6efd2.jpg)
![Screenshot 2021-05-12 at 12 14 47](https://user-images.githubusercontent.com/1336821/117966587-fae91400-b31b-11eb-9ad8-0505294ea29b.jpg)
![Screenshot 2021-05-12 at 12 25 27](https://user-images.githubusercontent.com/1336821/117967641-2fa99b00-b31d-11eb-80c6-93decebf5952.jpg)

**Comment by dom-whooley** [comment]
I will check but I think the last Sign and Register tabs with the white highlight and darker blue background can be removed.

**Comment by i-hardy** [comment]
I think it's important to draw a distinction between this kind of style for navigation links- we also have them on [support.theguardian.com](https://support.theguardian.com/uk/support)- or section headings, and tabs/tab panels as an interface, which refers to [a specific kind of interactive component](https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel). We have an example of the latter on the [paper subscriptions landing page](https://support.theguardian.com/uk/subscribe/paper).

The sign in/register example looks like it's actually a tabbed interface, but I think calling the other examples 'tabs' is a little misleading, as frequently they're just regular links or it's just a visual style around a heading.

**Reply by oliverlloyd** [reply]
This is an excellent point @i-hardy. It really does look like there are two separate things here, the Tab as a navigation tool and the 'Tab' as a heading. The last example with Sign In + Register side by side made me conflate them but based on what @dom-whooley said, that example is no longer valid.

I also agree with you that in this case, where there is no navigational aspect, then 'Tab' as the name is misleading.

I wonder if reducing scope like this leaves us in a position where the case for this as a component is not so strong? I would still like to see this heading text have consistent height and padding over multiple pages though, but perhaps the solution is a shared `SubHeader` component instead?

**Reply by dom-whooley** [reply]
Agreed, the first 3 are subheaders, of which a component could be made. The following 2 are navigations. The last one only exists as a design only, which is in the process of being changed to align with the first navigation version (yellow highlight)
